apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/KotlinCodegenLayout.kt

```diff
@@ -0,0 +1,17 @@
+package com.apollographql.apollo3.compiler.codegen.kotlin
+
+import com.apollographql.apollo3.compiler.PackageNameGenerator
+import com.apollographql.apollo3.compiler.codegen.CodegenLayout
+import com.apollographql.apollo3.compiler.escapeKotlinReservedEnumValueNames
+import com.apollographql.apollo3.compiler.escapeKotlinReservedWord
+
+class KotlinCodegenLayout(
+    packageNameGenerator: PackageNameGenerator,
+    schemaPackageName: String,
+    useSemanticNaming: Boolean,
+) : CodegenLayout(packageNameGenerator, schemaPackageName, useSemanticNaming) {
+
+  override fun escapeReservedWord(word: String): String = word.escapeKotlinReservedWord()
+
+  internal fun sealedClassValueName(name: String) = name.escapeKotlinReservedEnumValueNames()
```

Comment:
> The logic is inverted here
You're right 🤦.  In e3c7de8ddada712c6aaafb3b5aafb66a635bd61d I've renamed and added comments to make this absolutely less confusing

About `escapeReservedWord`: we still need to call `escapeKotlinReservedWord` because of `yield` that we escape ourselves

Reply:
> About escapeReservedWord: we still need to call escapeKotlinReservedWord because of yield that we escape ourselves

Right 👍  So it's more 3 kind of escapes

* underscore escapes
* apollo backtick escapes
* kotlinpoet backtick escapes

pheewww